rpcserver.go

```diff
@@ -3539,6 +3541,11 @@ func handleGetCoinSupply(s *rpcServer, cmd interface{}, closeChan <-chan struct{
 			tax = tax * voters / int64(params.TicketsPerBlock)
 			supply += (work + stake + tax)
 		}
+		isValid = dcrutil.IsFlagSet16(bh.VoteBits, dcrutil.BlockValid)
+
+		if i%int(params.ReductionInterval) == 0 {
+			base = base * params.DivSubsidy / params.MulSubsidy
```

Comment:
Like I said above, the blocks loop is reversed. So we need to run this backwards i.e increment it every `ReductionInterval` blocks until it reaches ~31.28DCR in the initial block.